pallets/parachain-staking/src/lib.rs

```diff
@@ -438,9 +445,51 @@ pub mod pallet {
 				weight = weight.saturating_add(T::OnNewRound::on_new_round(round.current));
 				// pay all stakers for T::RewardPaymentDelay rounds ago
 				weight = weight.saturating_add(Self::prepare_staking_payouts(round.current));
+
+				// select total collators
+				let collators = <SelectedCandidates<T>>::get();
+
+				// if collators length is below or eq to 66% of max_collators,
+				// we don't mark any other collator as offline
+				let max_collators = <TotalSelected<T>>::get();
+
+				// track collators length
+				let mut len_counter = collators.len();
+
+				// iter collators to check which of them must be marked as offline
+				for collator in collators {
+					if let Some(max_offline_counter) = <CandidateLastActive<T>>::get(&collator) {
+						if len_counter * 3 > (max_collators * 2) as usize
+							&& max_offline_counter >= T::MaxOfflineRounds::get()
```

Comment:
Optimization suggestion:  check this condition before the other one, because it's faster to compute

Reply:
I mean, replace `A && B` by `B && A`